Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/TicketService.hs

```diff
@@ -594,3 +674,244 @@ postTicketBookingsUpdateSeats _ TicketBookingUpdateSeatsReq {..} = do
               }
       QTSM.create seatM
   pure Kernel.Types.APISuccess.Success
+
+postTicketServiceCancel :: (Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person), Kernel.Types.Id.Id Domain.Types.Merchant.Merchant) -> API.Types.UI.TicketService.TicketServiceCancelReq -> Environment.Flow Kernel.Types.APISuccess.APISuccess
+postTicketServiceCancel (_mbPersonId, merchantId) req = do
+  now <- getCurrentTime
+  fork "cancel and refund:" $ do
+    preventFutureTicketBookingServiceStatus req.date req.description Domain.Types.SpecialOccasion.Closed req.ticketServiceCategoryId merchantId
+    bHoursRes <- QBH.findById req.businessHourId >>= fromMaybeM (BusinessHourNotFound req.businessHourId.getId)
+
+    ticketBookings <- QTB.getAllBookingsByPlaceIdAndVisitDate req.placeId req.date DTTB.Booked
```

Comment:
Let's first fetch `ticketBookingServices` and fetch `ticketBookings` from there. That way, you'll only have bookings for that particular business hour

Reply:
yeah but doing so will we will have to fetch from a list of ticket Booking that may increase time

Reply:
Currently you are fetching all the bookings for a particular day. Using this approach you will only fetch for a particular business hour on that day which will be smaller than this